compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -2187,8 +2187,12 @@ impl<'tcx> TyCtxt<'tcx> {
     ) -> Option<ImplOverlapKind> {
         // If either trait impl references an error, they're allowed to overlap,
         // as one of them essentially doesn't exist.
-        if self.impl_trait_ref(def_id1).map_or(false, |tr| tr.references_error())
-            || self.impl_trait_ref(def_id2).map_or(false, |tr| tr.references_error())
+        if self
+            .bound_impl_trait_ref(def_id1)
+            .map_or(false, |tr| tr.skip_binder().references_error())
```

Comment:
`subst_identity` here. We check whether the trait ref in it's own context already references an error.

tbh we're actually checking whether the bound trait ref references an error but can't just do `tr.references_error()` because `EarlyBinder` doesn't allow that to prevent users from accidentally doing that.